src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+
+using Microsoft.PowerShell.Commands;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Internal.Json
+{
+    /// <summary>
+    /// Class that defines Dsc cache entries.
+    /// </summary>
+    internal class DscClassCacheEntry
+    {
+        /// <summary>
+        /// Store the RunAs Credentials that this DSC resource will use.
+        /// </summary>
+        public DSCResourceRunAsCredential DscResRunAsCred { get; set; }
+
+        /// <summary>
+        /// If we have implicitly imported this resource, we will set this field to true. This will
+        /// only happen to InBox resources.
+        /// </summary>
+        public bool IsImportedImplicitly { get; set; }
+
+        /// <summary>
+        /// A CimClass instance for this resource.
+        /// </summary>
+        public PSObject CimClassInstance { get; set; }
+
+        /// <summary>
+        /// Initializes variables with default values.
+        /// </summary>
+        public DscClassCacheEntry() : this(DSCResourceRunAsCredential.Default, false, null) { }
```

Comment:
```suggestion
        /// <summary>
        /// Initializes variables with default values.
        /// </summary>
        public DscClassCacheEntry()
            : this(DSCResourceRunAsCredential.Default, aIsImportedImplicitly: false, aCimClassInstance: null)
        {
        }
```

Reply:
Also, I think our usual convention is to put constructors above properties

Reply:
updated.